### ISSUE TITLE
unnecessary_get_then_check miscategorised as "suspicious", should be "complexity"

### ISSUE THREAD

**Issue by ijackson** [rust-lang/rust-clippy#16990]
### Summary

In my project I turn off the "complexity" category.  I still want to get "suspicious", which warns on patterns that look like they might be bugs.

 `unnecessary_get_then_check` still triggers for me.  I don't think the `.get(...).is_some()` pattern is "suspicious".  It is more wordy than perhaps needed, but there is no reason to think it doesn't do what the programmer intended.

The suggestion, to replace `.get().is_some()` with `.contains()`, demonstrates that this fits precisely into the "complexity" category, about which the manual says "suggestions on how to simplify your code. It mostly focuses on code that can be written in a shorter and more readable way, while preserving the semantics.".

(I note that the description in the manual fof "suspicous" is very vague.  "code that is really sus" ?!)

### Lint Name

unnecessary_get_then_check

### Reproducer

I tried this code:

```rust
#![allow(clippy::style)]
#![allow(clippy::complexity)]

use std::collections::HashMap;

fn main() {
    let m = [(1,2)].into_iter().collect::<HashMap<_, _>>();
    println!("{:?}", m.get(&1).is_some());
}
```

I saw this happen:

```
warning: unnecessary use of `get(&1).is_some()`
 --> src/main.rs:8:24
  |
8 |     println!("{:?}", m.get(&1).is_some());
  |                        ^^^^^^^^^^^^^^^^^ help: replace it with: `contains_key(&1)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.95.0/index.html#unnecessary_get_then_check
  = note: `#[warn(clippy::unnecessary_get_then_check)]` on by default
```

I expected to see this happen:

No complaints.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=e8afaca1f15448ad9f27addce75d6aa8

### Version

```text
rustc 1.94.0-beta.2 (23a44d3c7 2026-01-25)
binary: rustc
commit-hash: 23a44d3c70448c08dc6a2fc13c1afceab49f2bb9
commit-date: 2026-01-25
host: x86_64-unknown-linux-gnu
release: 1.94.0-beta.2
LLVM version: 21.1.8
```
